### ISSUE TITLE
CMD+P to change focus on selected file pane (instead of opening again in selected pane)

### ISSUE THREAD

**Issue by krstp** [zed-industries/community#2231]
### Check for existing issues

- [X] Completed

### Describe the feature

Currently when `CMD+P` is used the selected file opens in the pane within which the `CMD+P` is invoked. This makes sense if the file was not opened. However, whenever the file is opened already within one of the panes, it makes sense the `CMD+P` selection would change the focus to that pane. Currently, this is not the case.

In summary: I would love `CMD+P` file selection to change the pane focus (where the file is already opened) instead of opening the file again in the pane of `CMD+P` invocation.

**Comment by gldtn** [comment]
For me It opens a new split tab if it's the same file you're currently focused on which makes sense, and changes focus if you don't have that file focused. Perhaps it would be nice to change the order and have the focused file at the bottom so we don't have to hit cmd-p cmd-p twice to change focus.

Or have it as the second item if we have more than 2 tabs open so we can quickly switch back and fourth between previews tab focused.

**Comment by JosephTLyons** [comment]
> For me It opens a new split tab if it's the same file you're currently focused on which makes sense, and changes focus if you don't have that file focused. Perhaps it would be nice to change the order and have the focused file at the bottom so we don't have to hit cmd-p cmd-p twice to change focus.
> 
> Or have it as the second item if we have more than 2 tabs open so we can quickly switch back and fourth between previous focused tab.

I think it is only supposed to open it in a second split if you are holding down `cmd` when you hit `enter` on the selection. Is Zed opening the new file in a second split, even if you are not holding down `cmd` when hitting `enter` on the file you want to select?

**Comment by krstp** [comment]
Correct. Currently, regardless of action, it will always open in new split, unless the focus is on the in-works pane. To be more precise: try to have 3, 4 panes open… with one of the action it will split the active pane and in the other case it will just create additional pane out of the last pane. So it seems to me it needs some polishing. Thanks for following up.

**Comment by gldtn** [comment]
> > For me It opens a new split tab if it's the same file you're currently focused on which makes sense, and changes focus if you don't have that file focused. Perhaps it would be nice to change the order and have the focused file at the bottom so we don't have to hit cmd-p cmd-p twice to change focus.
> > Or have it as the second item if we have more than 2 tabs open so we can quickly switch back and fourth between previous focused tab.
> 
> I think it is only supposed to open it in a second split if you are holding down `cmd` when you hit `enter` on the selection. Is Zed opening the new file in a second split, even if you are not holding down `cmd` when hitting `enter` on the file you want to select?

Aha, yes I was holding `cmd` while hitting enter, it's working as you described @JosephTLyons . Sorry for hijacking @krstp 

Just for clarification, I'm on Preview release 0.111.4 and it seems like it's working as intended, which version are you running @krstp ?

**Comment by krstp** [comment]
Just updated to `0.111.4`. Maybe there is misunderstanding how we interpret what we would expect and what happens.

1. After pushing `CMD+P` + `Enter` I would expect ZED to move to the pane where the selected file is located (currently this is not the case, unless condition 4 is met).
2. Currently, when `CMD+P` + `Enter` is selected it would open the file (again) in the currently active pane (Note: active pane is not the pane where the file is/was initially opened).
3. `CMD+P` + `CMD+Enter` indeed will open the file in a new split pane.
4. If the active pane already has one of the tabs with the file opened,  when I select (`CMD+P` + `Enter) it will refocus to the tab with the already opened tab.

Now: what I am trying to accomplish is point `1`.  `1` holds true if condition `4` is met. However, when condition `2` is met, it does not fulfill condition `1`. Condition `3` works as epxected.

I hope this clarifies. If this is still not clear I can make a video later.

**Comment by JosephTLyons** [comment]
Sorry for the confusion here. I think I understand both @krstp's original request and also what was going on with what @gldtn - they aren't the same thing. @krstp wants for zed to know that a file is already open in another pane and to focus on it, even if the pane it is in isn't the active one - they don't want zed to keep opening new tabs in the active pane if it exists somewhere else in another pane. @gldtn was experiencing something different - that picking a file in the file finder always opened it in a new pane, because they were holding down the `cmd` button when selecting the file in the picker. Just wanted to iterate that I *think* I understand the original request and that confusion might've happened when I was explaining to @gldtn what might be going on for them.
